### ISSUE TITLE
Compound assignment is not supported.

### ISSUE THREAD

**Issue by holdenmai** [dynamicexpresso/DynamicExpresso#251]
Currently, the below is not supported.
```

			var target = new Interpreter();

			var parameters = new[]{
							new Parameter(nameof(a), typeof(int)),
							new Parameter(nameof(b), typeof(int)),
							};
			Assert.DoesNotThrow(() => target.Parse("a += b", parameters));
```

As per the [operator overloading](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#114-operators) documentation, "When a binary operator is overloaded, the corresponding compound assignment operator, if any, is also implicitly overloaded."

This can be worked around by changing our eval text to 
`"a = (a + b)"`

**Comment by davideicardi** [comment]
Another consideration is that allowing to change parameter value can cause some problems (security?). Consider the case where where two expressions are evaluated with the same parameter and the first expression change it ...
I know that from C# perspective this is valid, but maybe considering parameters read only is better in term of design.

**Comment by holdenmai** [comment]
As an example where this can already be a valid use case from one of the existing unit tests (Assignment_Operator_Equal) would be to have an additional test like (to show a lot of edge cases)

```
x.Property1 = 4;
x.Property2 = 2;
target.Eval("x.Property1 += x.Property2 *= 10");
Assert.AreEqual(20, x.Property2);
Assert.AreEqual(24, x.Property1);
```

**Comment by davideicardi** [comment]
Yes, I agree with you that this can be a complex discussion. In the past I have implemented this flag to disable assignements:
```
new Interpreter().EnableAssignment(AssignmentOperators.None);
```
Eventually we can do something similar ...
